navigator/androidx-nav/src/main/java/com/freeletics/mad/navigator/internal/AndroidXDeepLinkHandler.kt

```diff
@@ -0,0 +1,65 @@
+package com.freeletics.mad.navigator.internal
+
+import android.app.Activity
+import android.content.Intent
+import android.os.Bundle
+import androidx.navigation.NavController.Companion.KEY_DEEP_LINK_ARGS
+import androidx.navigation.NavController.Companion.KEY_DEEP_LINK_IDS
+import com.eygraber.uri.Uri
+import com.freeletics.mad.navigator.BaseRoute
+import com.freeletics.mad.navigator.DeepLink
+import com.freeletics.mad.navigator.DeepLink.Companion.EXTRA_DEEPLINK_ROUTES
+import com.freeletics.mad.navigator.DeepLinkHandler
+
+/**
+ * Handles 2 different kind of deep links:
+ * - url based deep links when the `Activity` was launched with a standard `ACTION_VIEW` `Intent`
+ *   that contains the url in [Intent.getData]
+ * - programmatically launched deep links when the `Activity` was launched with an `Intent` created
+ *   through one of the `build...` methods on [DeepLink]
+ *
+ * In the case of url based deep links the given [deepLinkHandlers] will be searched for a match
+ * through which a [DeepLink] will be created. An `Intent` created with [DeepLink.buildIntent] will
+ * then replace the original [Activity.getIntent].
+ *
+ * For programmatically launched deep links the current `Intent` will be augmented with additional
+ * information for AndroidX.
+ */
+@InternalNavigatorApi
+public fun Activity.handleDeepLink(
```

Comment:
This is basically a replacement of our internal `DeepLinkDispatcher` but for single activity apps. It also takes care of converting the routes into a format that AndroidX can handle which was previously directly inside `DeepLink.buildIntent`